File: javascript/src/models/endpointAttemptStats.ts
// this file is @generated

export interface EndpointAttemptStats {
  success: number;
  fail: number;
  canceled: number;
}

export const EndpointAttemptStatsSerializer = {
  _fromJsonObject(object: any): EndpointAttemptStats {
    return {
      success: object["success"],
      fail: object["fail"],
      canceled: object["canceled"],
    };
  },

  _toJsonObject(self: EndpointAttemptStats): any {
    return {
      success: self.success,
      fail: self.fail,
      canceled: self.canceled,
    };
  },
};

File: rust/src/models/endpoint_attempt_stats.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct EndpointAttemptStats {
    pub success: u64,

    pub fail: u64,

    pub canceled: u64,
}

impl EndpointAttemptStats {
    pub fn new(success: u64, fail: u64, canceled: u64) -> Self {
        Self {
            success,
            fail,
            canceled,
        }
    }
}
